.evergreen/shared.yml

```diff
@@ -0,0 +1,408 @@
+      params:
+        binary: bash
+        args:
+          - ui/${app_dir}/scripts/prepare-shell.sh
+    - command: expansions.update
+      params:
+        file: expansion.yml
+    - command: shell.exec
+      params:
+        shell: bash
+        script: |
+          ${PREPARE_SHELL}
+
+          # Fetch NVM and install it into this task's .nvm directory
+          # Once downloaded, source nvm and install yarn
+          git clone https://github.com/nvm-sh/nvm.git "$NVM_DIR"
+          cd "$NVM_DIR"
+          git checkout `git describe --abbrev=0 --tags --match "v[0-9]*" $(git rev-list --tags --max-count=1)`
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+          cd -
+
+          # Retry the download for Node in case it flakes.
+          for i in {1..5}; do
+            nvm install --no-progress --default ${node_version}
+            [[ $? -eq 0 ]] && break
+            echo "Attempt $i of 5 to install Node failed"
+            sleep 10
+          done
+          npm install -g yarn
+
+  symlink:
+    command: shell.exec
+    params:
+      working_dir: ui
+      shell: bash
+      script: |
+        ln -s ${workdir}/ui/evergreen/graphql/schema ${app_dir}/sdlschema
+
+  wait-for-evergreen:
+    command: shell.exec
+    type: setup
+    params:
+      exec_timeout_secs: 600
+      working_dir: ui/${app_dir}
+      script: ./scripts/wait-for-evergreen.sh
+
+  yarn-build:
+    command: shell.exec
+    params:
+      working_dir: ui/${app_dir}
+      shell: bash
+      script: |
+        ${PREPARE_SHELL}
+        yarn build:local
+
+  yarn-build-storybook:
+    command: shell.exec
+    params:
+      working_dir: ui/${app_dir}
+      script: |
+        ${PREPARE_SHELL}
+        # TODO: Standardize Storybook script name in Spruce and Parsley
+        yarn build-storybook || yarn storybook:build
+
+  yarn-cypress:
+    command: shell.exec
+    params:
+      working_dir: ui/${app_dir}
+      shell: bash
+      script: |
+        ${PREPARE_SHELL}
+        # Allow spec filtering for an intentional patch.
+        if [[ "${requester}" == "patch" ]]; then
+          yarn cy:run --reporter junit --spec "${cypress_spec}"
+        else
+          yarn cy:run --reporter junit
+        fi
+
+  yarn-eslint:
+    command: shell.exec
+    params:
+      working_dir: ui/${app_dir}
+      shell: bash
+      script: |
+        ${PREPARE_SHELL}
+        yarn eslint:strict
+
+  yarn-install:
+    command: shell.exec
+    params:
+      working_dir: ui/${app_dir}
+      shell: bash
+      script: |
+        ${PREPARE_SHELL}
+        yarn install
+
+  yarn-preview:
+    command: shell.exec
+    params:
+      working_dir: ui/${app_dir}
+      background: true
+      shell: bash
+      script: |
+        ${PREPARE_SHELL}
+        yarn preview
+
+  yarn-serve:
+    command: shell.exec
+    type: setup
+    params:
+      working_dir: ui/${app_dir}
+      background: true
+      script: |
+        ${PREPARE_SHELL}
+        yarn serve
```

Comment:
I would love if we could eventually consolidate these I always get confused when jumping between the repos on which one i should run

Reply:
DEVPROD-5869